test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -20,6 +20,38 @@ public void StringSegment_StringCtor_AllowsNullBuffers()
             Assert.Equal(0, segment.Length);
         }
 
+        [Fact]
+        public void StringSegmentConstructor_NullBuffer_Throws()
+        {
+            // Arrange, Act and Assert
+            Assert.Throws<ArgumentNullException>(() => new StringSegment(null, 0, 0));
```

Comment:
Can you `Assert` the param name so we know which one throws? Particularly useful in the new scenario